### PR TITLE
[CLC-502] v5.3.7 Release notes and improved documentation

### PR DESCRIPTION
**Pull request by yuce**

### CODE REVIEW

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/clc-project.adoc [hunk 1]
```diff
@@ -129,6 +129,50 @@ Item 1 = bar
 Item 2 = 38
 -----
 
+==== Custom Template Actions
+
+You can include the `template.star` script in the root of your project template to add your own actions to the templates.
+`template.star` file is used only during template generation, it will not be copied to the final project.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-project.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/clc-project.adoc [hunk 2]
```diff
@@ -129,6 +129,50 @@ Item 1 = bar
...
+
+Each function in the `template.star` script are converted to actions, except the ones that start with the underscore (`_`) character.
+The functions may take 0 or more characters but they should return a single value.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-project.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/clc-project.adoc [hunk 3]
```diff
@@ -129,6 +129,50 @@ Item 1 = bar
...
+[source]
+----
+{{ $myvar := 38 }}
+Result of concat: {{ concat "a string" $myvar }}
+----
+
+==== Success Note
+
+You can display a success note when a project is created successfully.
+This note may be useful to direct the user to the next step for using the project.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-project.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/clc-project.adoc [hunk 4]
```diff
@@ -129,6 +129,50 @@ Item 1 = bar
...
+You can use one or many of the following sections in `template.yaml`:
+
+* `note-unix`: Text to be displayed only on Linux or macOS.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-project.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/clc-project.adoc [hunk 5]
```diff
@@ -129,6 +129,50 @@ Item 1 = bar
...
+* `note-windows`: Text to be displayed only on Windows.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-project.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 6]
```diff
@@ -12,17 +12,47 @@
 == Installing on macOS
 
 The Hazelcast CLC is supported on macOS 15 or newer versions.
+You can install CLC using either by using an installation script or downloading the binary for your platform manually.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 7]
```diff
@@ -12,17 +12,47 @@
...
+
+=== Using the Install Script
+
+Run the following command to install CLC using the install script we provide:
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 8]
```diff
@@ -12,17 +12,47 @@
...
-+
 [source,shell,subs="attributes"]
 ----
 curl https://hazelcast.com/clc/install.sh | bash
 ----
 
+=== Manual Installation
+
+You can download a CLC package manually from our link:https://github.com/hazelcast/hazelcast-commandline-client/releases/latest[Releases] page.
+MacOS compatible packages are named in the `hazelcast-clc-VERSION_darwin_ARCHITECTURE` form.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 9]
```diff
@@ -12,17 +12,47 @@
...
+Mind the architecture that corresponds to your computer.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 10]
```diff
@@ -12,17 +12,47 @@
...
+
+You must extract the package once the download is complete.
+Extracting the package will create a folder which we will refer as `CLC_FOLDER` below.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 11]
```diff
@@ -12,17 +12,47 @@
...
+
+On macOS, binaries downloaded outside of AppStore require your intervention to run. The install script automatically handles this, but if you downloaded a release package you can do it manually:
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 12]
```diff
@@ -12,17 +12,47 @@
...
+
+[source,bash]
+----
+$ xattr -d com.apple.quarantine CLC_FOLDER/clc
+----
+Use the correct path instead of `CLC_FOLDER` in the command above.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 13]
```diff
@@ -12,17 +12,47 @@
...
+[source,bash]
+----
+$ sudo cp CLC_FOLDER/clc /usr/local/bin
+----
+
 == Installing on Linux
 
-The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22.04.
+The Hazelcast CLC runs on any recent Linux distribution.
+You can install CLC using either by using an installation script or downloading the binary for your platform manually.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 14]
```diff
@@ -31,10 +61,42 @@ The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22
 curl https://hazelcast.com/clc/install.sh | bash
 ----
 
+=== Manual Installation
+
+You can download a CLC package manually from our link:https://github.com/hazelcast/hazelcast-commandline-client/releases/latest[Releases] page.
+Linux compatible packages are named in the `hazelcast-clc-VERSION_linux_ARCHITECTURE` form.
+Mind the architecture that corresponds to your computer.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 15]
```diff
@@ -31,10 +61,42 @@ The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22
...
+
+You must extract the package once the download is complete.
+Extracting the package will create a folder which we will refer as `CLC_FOLDER` below.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 16]
```diff
@@ -31,10 +61,42 @@ The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22
...
+
+You may prefer to copy `clc` to a directory in your `$PATH`.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 17]
```diff
@@ -31,10 +61,42 @@ The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22
...
+
+
 == Installing on Windows
 
 The Hazelcast CLC is supported on Windows 10 or newer versions. We provide pre-built binaries only for 64bit Intel/AMD architecture.
 
+=== Manual Installation
+
+You can download a CLC package manually from our link:https://github.com/hazelcast/hazelcast-commandline-client/releases/latest[Releases] page.
+Linux compatible packages are named in the `hazelcast-clc-VERSION_windows_ARCHITECTURE` form.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 18]
```diff
@@ -31,10 +61,42 @@ The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22
...
+Mind the architecture that corresponds to your computer.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 19]
```diff
@@ -31,10 +61,42 @@ The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22
...
+
+You must extract the package once the download is complete.
+Extracting the package will create a folder which we will refer as `CLC_FOLDER` below.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 20]
```diff
@@ -31,10 +61,42 @@ The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22
...
+
+You may prefer to copy `CLC_FOLDER/clc` in a folder specified in the `%PATH%` environment variable.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/install-clc.adoc [hunk 21]
```diff
@@ -31,10 +61,42 @@ The Hazelcast CLC runs on any recent Linux distribution. We test it on Ubuntu 22
...
+Otherwise add `CLC_FOLDER/clc` to your `%PATH`.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/install-clc.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/release-notes-5.3.7.adoc [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+
+We have added the `clc config import-java` command that imports a subset of the Java client configuration and associated keystore/trustore files.
+
+Check out the documentation for the <<clc-config.adoc#clc-config-import-java, clc config import-java>> command for more information about this command.
+
+=== Project Creator
+
+==== Custom Template Actions
+
+It is now possible to define custom template actions using advanced scripts.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/release-notes-5.3.7.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/release-notes-5.3.7.adoc [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+
+Check out the <<clc-project.adoc#custom-template-actions, Custom Template Actions>> section for more information about this feature.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/release-notes-5.3.7.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/release-notes-5.3.7.adoc [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+
+==== Success Note
+
+You can add a success note when a project is created successfully.
+
+Check out the <<clc-project.adoc#success-note, Success Note>> section for more information about this command.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/release-notes-5.3.7.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/release-notes-5.3.7.adoc [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+
+== Changes
+
+* Viridian branding is updated to use Hazelcast Cloud.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/release-notes-5.3.7.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

File: docs/modules/ROOT/pages/release-notes-5.3.7.adoc [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+* `clc viridian` commands are renamed to `clc cloud`. `clc viridian` is set an alias to `clc cloud`.
+* `clc config import` command is renamed to `clc config import-cloud`. `clc config import` is an alias to `clc config import-cloud`.
+* `map unlock` was accidentally introduced to the non-interactive mode in the previous version. It is changed to be available to interactive/scripting modes only.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/release-notes-5.3.7.adoc]: Updated at: https://github.com/hazelcast/clc-docs/pull/17/commits/4a6cf6c4e8b8b27c5a75297cdedc0751be68a88d

**Comment by yuce** [comment]
@rebekah-lawrence Thanks for the great review! I've applied your suggestions.

**Review by rebekah-lawrence — APPROVED** [review]
LGTM

My pleasure, @yuce - really appreciate the work you put into these docs!

**Comment by yuce** [comment]
@rebekah-lawrence The documentation I write would be a mess without your improvements. Thanks a lot! :)

**Pull request merged**